components/pd_client/src/client.rs

```diff
@@ -73,6 +76,9 @@ impl RpcClient {
                             client,
                             members,
                         )),
+                        monitor: yatp::Builder::new(thd_name!("pdmonitor"))
+                            .max_thread_count(1)
+                            .build_future_pool(),
```

Comment:
How about building the thread pool outside the loop, so we don't need to build it on every retry.